### PR TITLE
fix some style / warnings

### PR DESCRIPTION
**Pull request by joevandyk**
not sure if these sorts of things are valuable.. I like to try to follow the ruby style guide when possible. does dry-rb have a defined style guide?

### CODE REVIEW

**Comment by fran-worley** [comment]
Is there a reason why you've replaced all cases of .() with .call() ?

**Comment by joevandyk** [comment]
https://github.com/bbatsov/ruby-style-guide#proc-call

**Comment by fran-worley** [comment]
I'm happy with all the changes except that one. 

If you replace `.()` with `.call()` here then for consistency you ought to do it in the other libs. I get that standards are supposed to replace personal preferences but now I get what `.()` does I'm quite attached. 

@solnic what are your thoughts?

**Comment by fran-worley** [comment]
@joevandyk I think you've got `raise` and `fail` the wrong way around: https://github.com/bbatsov/ruby-style-guide#exceptions

Looks like it should be raise over fail (i.e. how it was).

File: lib/dry/logic/rule.rb [hunk 1]
```diff
@@ -48,8 +48,8 @@ def new(predicate)
         self.class.new(predicate, options)
       end
 
-      #bit of a hack, essentially genuine proc.curry should be provided it's args via[] whereas ours
-      #accepts them as a method param
+      # bit of a hack, essentially genuine proc.curry should be provided it's args via[] whereas ours
```
Comment (fran-worley) [inline lib/dry/logic/rule.rb]: This comment actually relates to the private `curried_predicate` method. Can you move it?

**Comment by solnic** [comment]
Please don't change `.()` to `call`, I like to use it in places where it's a call w/o side-effects, and it's shorter.

File: lib/dry/logic/predicate.rb [hunk 2]
```diff
@@ -4,17 +4,17 @@ def self.Predicate(block)
...
 
     class Predicate
-      Undefined = Class.new {
+      Undefined = Class.new do
         def inspect
           "undefined"
         end
         alias_method :to_s, :inspect
-      }.new.freeze
+      end.new.freeze
```
Comment (solnic) [inline lib/dry/logic/predicate.rb]: Please change it back to `{ .. }`, `do .. end` should be used in places where return value is not used, `end.new` is really ugly IMHO

**Comment by solnic** [comment]
We use rubocop on CodeClimate but there's no official dry-rb style guide (yet). We definitely use a slightly different style so I guess we should document it somewhere. I'm happy to merge this in assuming you're OK with fixing the things I mentioned in my comments :) Thank you.

**Comment by joevandyk** [comment]
I'll close this.. perhaps there could be an official dry-rb rubocop config shared between the projects?

**Comment by solnic** [comment]
We can think about a common config and using either CodeClimate or rubocop on CI but I wouldn't want to maintain it, so 👎 unless we have people who are willing to do this kind of work.

**Pull request closed without merging**